xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DocumentsMacro.xml

```diff
@@ -307,11 +307,13 @@ Generates:
   #set ($limit = $services.rendering.escape($xcontext.macro.params.count, 'xwiki/2.1'))
 #end
 #set ($propertiesStr = $services.rendering.escape($stringtool.join($properties, ','), 'xwiki/2.1'))
+#set ($filters = $xcontext.macro.params.filters)
 {{liveData
   id="$livedataId"
   properties="$propertiesStr"
   source="liveTable"
   sourceParameters="$escapetool.url($sourceParameters)"
+  #if ($filters) filters="$filters"#end
```

Comment:
This needs escaping, otherwise this allows remote code execution.

Reply:
fixed by https://github.com/xwiki/xwiki-platform/pull/4153/files#diff-ff93f488ff7153ca0b24a66833bbbe857b775250589771d300a0fd2780ab2efeR310

Reply:
This if condition is good enough here? Don't you need a `if ("$!filters" != '')`, or do you need the if at all (i.e. what happens if you put `filters=""` in the macro?

Reply:
You are right, to be strict we should add a parameter to the LD macro if the `filters` param is explicitly passed to the documents macro.

Reply:
Addressed by https://github.com/xwiki/xwiki-platform/pull/4153/files#diff-ff93f488ff7153ca0b24a66833bbbe857b775250589771d300a0fd2780ab2efeR310